Dockerfile

```diff
@@ -39,31 +36,158 @@ RUN conda install --name python2 --quiet --yes \
 # https://github.com/damianavila/RISE
 RUN conda install --quiet --yes -c damianavila82 rise
 
+# install zeroc-ice and python-omero
 RUN conda install --name python2 --quiet --yes -c bioconda zeroc-ice && \
-    conda install --name python2 --quiet --yes -c pdrops pygraphviz && \
-    /opt/conda/envs/python2/bin/pip install \
+    conda install --name python2 --quiet --yes -c bioconda python-omero=5.3.3
+
+# install idr-py and notebook dependencies
+RUN /opt/conda/envs/python2/bin/pip install \
         graphviz \
         gseapy \
         py2cytoscape \
         pydot \
         tqdm \
-        idr-py==0.1
-
-# Display resource usage in notebooks https://github.com/yuvipanda/nbresuse
-RUN pip install https://github.com/IDR/nbresuse/archive/0.1.0-idr.zip && \
-    jupyter serverextension enable --py nbresuse && \
-    jupyter nbextension install --py --user nbresuse && \
-    jupyter nbextension enable --py --user nbresuse
+        idr-py==0.1.1
 
 RUN mkdir -p /home/jovyan/.local/share/jupyter/kernels/python2 && \
     sed 's/Python 2/OMERO Python 2/' \
         /opt/conda/envs/python2/share/jupyter/kernels/python2/kernel.json > \
         /home/jovyan/.local/share/jupyter/kernels/python2/kernel.json
 
-# Don't rename user- causes problems with earlier hardcoded paths
-#RUN usermod -l omero jovyan -m -d /home/omero
-#USER omero
+# Install git and pull the notebooks from the training repository
+RUN conda install --name python2 --quiet --yes -c anaconda git
+
+# Switch to root user for installing Cell Profiler dependencies
+USER root
+
+# Install CellProfiler dependencies
+RUN   apt-get -y update &&                                          \
+      apt-get -y install                                            \
+        build-essential    \
+        cython             \
+        git                \
+        libmysqlclient-dev \
+        libhdf5-dev        \
+        libxml2-dev        \
+        libxslt1-dev       \
+        openjdk-8-jdk      \
+        python-dev         \
+        python-pip         \
+        python-h5py        \
+        python-matplotlib  \
+        python-mysqldb     \
+        python-scipy       \
+        python-numpy       \
+        python-wxgtk3.0    \
+        python-zmq
+
+WORKDIR /usr/local/src
+
+# Install CellProfiler
```

Comment:
Is `/opt/conda/envs/python2/bin/pip install https://github.com/CellProfiler/CellProfiler/archive/v3.0.0.zip` good enough?

Reply:
I used the docker file snippet that they had shared, https://github.com/CellProfiler/docker. Not sure if the pip installable module takes care of all the dependencies.